### PR TITLE
Clean up and improve documentation of x86_32 registers

### PR DESCRIPTION
**Pull request by hainest**
This also fixes the bug that the MMX* and st* registers have identical values. This prevents having all of the registers in a single switch because duplicate literal case values are not allowed (e.g., `case Dyninst::x86::immx0` and `case Dyninst::x86::ist0`).

### CODE REVIEW

**Comment by hainest** [comment]
Testing fine on all platforms.

File: common/src/registers/MachRegister.C [hunk 1]
```diff
@@ -32,6 +32,14 @@ namespace Dyninst {
           return MachRegister(reg & 0xfffff0ff);
```
Comment (kupsch) [inline common/src/registers/MachRegister.C:32]: clear the whole 16-bits of the subrange with a mask of `0xffff00ff`.

**Review by kupsch — CHANGES_REQUESTED** [review]
A couple changes in getBaseRegister, then it looks good to me.

If you feel ambitious, you could make and use mask constants and for the components of a register (id, alias/subrange, category, arch), so it is clearer, and less implementation details are spilled everywhere. Even more ambitious would be to create and use functions to get the components of the register that use the masks.

File: common/h/registers/x86_regs.h [hunk 2]
```diff
@@ -38,42 +38,50 @@
...
+  const signed int FLAG   = 0x00030000;  // EFLAGS Register
+  const signed int MISC   = 0x00040000;  // Internal ProcControlAPI Register
+  const signed int CTL    = 0x00050000;  // Control Registers CR0-CR7
+  const signed int DBG    = 0x00060000;  // Debug Registers DR0-DR7
+  const signed int TST    = 0x00070000;  // Internal InstructionAPI Registers
+  const signed int X87    = 0x00080000;  // x87 FPU Registers
+  const signed int MMX    = 0x00090000;  // MM0-MM7 Registers
+  const signed int XMM    = 0x000A0000;  // XMM0-XMM7 Registers from SSE
+  const signed int YMM    = 0x000B0000;  // YMM0-YMM7 Registers from AVX2/FMA
+  const signed int ZMM    = 0x000C0000;  // ZMM0-ZMM7 Registers from AVX-512
```
[the remark this answers is not in the document]
Comment (hainest) [reply common/h/registers/x86_regs.h:81]: From the SDM:

```
15.1.2 32 SIMD Register Support
      Intel AVX-512 instructions also support 32 SIMD registers in 64-bit
      mode (XMM0-XMM31, YMM0-YMM31 and ZMM0-ZMM31). The number of available
      vector registers in 32-bit mode is still 8.
```

I had assumed the 8 vector registers are ZMM0-ZMM7, but maybe it's referring to YMM0-YMM7 (ZMM0-ZMM3)?

[the remark this answers is not in the document]
Comment (kupsch) [reply common/h/registers/x86_regs.h:81]: I thought I deleted this comment (it went away in github, but the lines were still highlighted).  ZMM0-7 can exist in x86-32.

**Review by kupsch — APPROVED** [review]

File: common/h/registers/x86_regs.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (hainest) [reply common/h/registers/x86_regs.h:81]: Github comments are... special. Good to see that I read that part correctly.

**Pull request merged**